flexmeasures/data/models/planning/tests/test_solver.py

```diff
@@ -792,3 +792,75 @@ def test_validate_constraints(
     )
 
     assert set(expected_constraint_type_violations) == constraint_type_violations_output
+
+
+def test_infeasible_problem_error(add_battery_assets):
+    """Try to create a schedule with infeasible constraints. soc-max is 4.5 and soc-target is 8.0"""
+
+    # get the sensors from the database
+    _epex_da, battery = get_sensors_from_db()
+
+    # time parameters
+    _tz, start, end, resolution = get_time_parameters(
+        [2015, 1, 2], [2015, 1, 3], {"hours": 1}
+    )
+
+    def compute_schedule(flex_model):
+        scheduler = StorageScheduler(
+            battery,
+            start,
+            end,
+            resolution,
+            flex_model=flex_model,
+        )
+        schedule = scheduler.compute()
+
+        soc_schedule = integrate_time_series(
+            schedule,
+            soc_at_start,
+            decimal_precision=1,
+        )
+
+        return soc_schedule
+
+    # soc parameters
+    soc_at_start = battery.get_attribute("soc_in_mwh")
+    soc_min = 0.5
+    soc_max = 4.5
+    soc_maxima = [{"datetime": "2015-01-02T15:00:00+01:00", "value": 1.0}]
+    soc_minima = [{"datetime": "2015-01-02T08:00:00+01:00", "value": 3.5}]
+
+    soc_targets = [{"datetime": "2015-01-02T16:00:00+01:00", "value": 8.0}]
+
+    flex_model = {
+        "soc-at-start": soc_at_start,
+        "soc-min": soc_min,
+        "soc-max": soc_max,
+        "soc-maxima": soc_maxima,
+        "soc-minima": soc_minima,
+        "soc-targets": soc_targets,
+    }
+
+    with pytest.raises(
+        ValueError, match="The input data yields an infeasible problem."
+    ):
+        compute_schedule(flex_model)
+
+
+def get_sensors_from_db():
+    # get the sensors from the database
+    epex_da = Sensor.query.filter(Sensor.name == "epex_da").one_or_none()
+    battery = Sensor.query.filter(Sensor.name == "Test battery").one_or_none()
+    assert battery.get_attribute("market_id") == epex_da.id
+
+    return epex_da, battery
+
+
+def get_time_parameters(start_date: list, end_date: list, resolution: dict):
```

Comment:
I don't see the benefit of having this function, the only thing is not to repeat the timezone localization of `start` and `end`. This function can be called later in other parts of the code and get the order of the day, month, year wrong.

I suggest keeping just the `start` end `end` as timezone unaware datetimes. In addition, accept the timezone as an optional string that is `Europe/Amsterdam`, by default.

Reply:
removed this refactor